pinot-core/src/test/java/org/apache/pinot/queries/JsonDataTypeQueriesTest.java

```diff
@@ -123,6 +118,14 @@ public void setUp()
         "{\"name\": {\"first\": \"goofy\", \"last\": \"dwag\"}, \"id\": 171, \"data\": [\"k\", \"b\", \"c\", \"d\"]}"));
     records.add(createRecord(9, "ludwik von drake",
         "{\"name\": {\"first\": \"ludwik\", \"last\": \"von drake\"}, \"id\": 181, \"data\": [\"l\", \"b\", \"c\", \"d\"]}"));
+    records.add(createRecord(10, "nested array",
+        "{\"name\":{\"first\":\"nested\",\"last\":\"array\"},\"id\":111,\"data\":[{\"e\":[{\"x\":[{\"i1\":1,\"i2\":2}]},{\"y\":[{\"i1\":1,\"i2\":2}]},{\"z\":[{\"i1\":1,\"i2\":2}]}]},{\"b\":[{\"x\":[{\"i1\":1,\"i2\":2}]},{\"y\":[{\"i1\":1,\"i2\":2}]},{\"z\":[{\"i1\":10,\"i2\":20}]}]}]}"));
```

Comment:
(nit)

If I were to write these test cases, I would add json formatted data structure of the record before each test. It will help with 2 things

- When reading the test data and corresponding test case, it really makes it more readable to understand the functionality and what is being test
- Tomorrow if someone has to debug this test, pretty printed json data will help with debugging. Single one line json string isn't super helpful and will only make it worse as more complicated test cases are added.

Will make the code verbose, but will be helpful imo

Reply:
Yeah, I was trying to make the code less verbose. I just copy and paste it into https://jsonlint.com/ in browser window for ease of viewing. I haven't tried it yet, but this looks useful too: https://plugins.jetbrains.com/plugin/9679-json-viewer.